analysis/import/SCMLogParserV2/src/main/kotlin/de/maibornwolff/codecharta/importer/scmlogparserv2/converter/ProjectConverter.kt

```diff
@@ -0,0 +1,78 @@
+package de.maibornwolff.codecharta.importer.scmlogparserv2.converter
+
+import de.maibornwolff.codecharta.importer.scmlogparserv2.input.VersionControlledFile
+import de.maibornwolff.codecharta.importer.scmlogparserv2.input.metrics.MetricsFactory
+import de.maibornwolff.codecharta.importer.scmlogparserv2.parser.VersionControlledFilesInGitProject
+import de.maibornwolff.codecharta.importer.scmlogparserv2.parser.VersionControlledFilesList
+import de.maibornwolff.codecharta.model.Edge
+import de.maibornwolff.codecharta.model.MutableNode
+import de.maibornwolff.codecharta.model.NodeType
+import de.maibornwolff.codecharta.model.PathFactory
+import de.maibornwolff.codecharta.model.Project
+import de.maibornwolff.codecharta.model.ProjectBuilder
+
+/**
+ * creates Projects from List of VersionControlledFiles
+ */
+class ProjectConverter(private val containsAuthors: Boolean) {
+
+    private val ROOT_PREFIX = "/root/"
+
+    private fun addVersionControlledFile(projectBuilder: ProjectBuilder, versionControlledFile: VersionControlledFile) {
+        val attributes = extractAttributes(versionControlledFile)
+        val edges = versionControlledFile.getEdgeList()
+        val fileName = versionControlledFile.filename.substringAfterLast(PATH_SEPARATOR)
+        val newNode = MutableNode(fileName, NodeType.File, attributes, "", mutableSetOf())
+        val path = PathFactory.fromFileSystemPath(
+            versionControlledFile.filename.substringBeforeLast(PATH_SEPARATOR, "")
+        )
+
+        projectBuilder.insertByPath(path, newNode)
+        edges.forEach { projectBuilder.insertEdge(addRootToEdgePaths(it)) }
+        versionControlledFile.removeMetricsToFreeMemory()
+    }
+
+    private fun extractAttributes(versionControlledFile: VersionControlledFile): Map<String, Any> {
+        return when {
+            containsAuthors ->
+                versionControlledFile.metricsMap
+                    .plus(Pair("authors", versionControlledFile.authors))
+            else -> versionControlledFile.metricsMap
+        }
+    }
+
+    private fun addRootToEdgePaths(edge: Edge): Edge {
```

Comment:
We should check, if the Edge object could do that prefixing itself.

Reply:
The Edge object is responsible for all edges in created by the SCMLogParser
Creating an extra function addRootEdge() and calling it in ProjectConverter would make it more confusing as the function would be callable at different code spots, while the functional itself is only needed once at project creation.

Therefore I think we should not extract it, or what would you reason?

Reply:
I thought that I don't want to take care of adding some prefixes in any place outside of the Edge object.
This should not be the job of the ProjectConverter, I guess. Rather the Edge object (https://github.com/MaibornWolff/codecharta/blob/feature/846/scmlogparser/analysis/model/src/main/kotlin/de/maibornwolff/codecharta/model/Edge.kt) should encapsulate this functionality or am I wrong?

Reply:
But that would make the function accessible from outside, cause the Edge object has no way of knowing if it is a root edge or not without additional changes.

Reply:
It looks like that all edges will be prefixed here. So my thought was that we can add the Root Prefix inside the Edge object for all edges. If I am wrong, let's skip this suggestion.